infrastructure/db/handler.js

```diff
@@ -1,253 +1,208 @@
+    models.teammember.create({ emailAddress : member.emailAddress, 
+        password : member.password });
 
 }
 
 /**
 * Finds all the Teammembers of specific board
 * @param {int} boardId id of the wanted board
-* @return {Promise} promise, with json response
+* @return {JSON} json, with board's teammembers details
 */
-exports.getBoardTeammembers = async(boardId) => {
-    
-    return new Promise((resolve) => {
-
-        models.board.findByPk(boardId).then((board) => {
-            board.getTeammembers().then(users => {
-
-                let resp = { Teammembers : new Array() };
-                for(let i = 0;i < users.length;i++) {
-                    resp.Teammembers[i] = {
-                        id : users[i].id,
-                        emailAddress : users[i].emailAddress,
-                        createdAt : users[i].createdAt,
-                        updatedAt : users[i].updatedAt
-                    }
-                }
-                resolve(resp);
-
-            });
-        });
-
-    });
+const getBoardTeammembers = async(boardId) => {
+    try {
+        const board = await getBoard(boardId);
+        if(board != null) {
+            return await board.getTeammembers();
+        } else {
+            return [];
+        }
+    } catch(err) {
+        throw Error(err);
+    }
 }
 /**
 * Finds all the Groups of specific board
 * @param {int} boardId id of the wanted board
-* @return {Promise} promise, with json response
+* @return {JSON} json with all groups of requested board
 */
-exports.getGroups = async(boardId) => {
-
-    return new Promise((resolve) => {
-
-        models.group.findAll({ where : { boardId : boardId } }).then(groups => {
-
-            let resp = { groups : new Array() }; //Builds array of all the groups
-            for(let i = 0;i < groups.length;i++) {
-                resp.groups[i] = {
-                    id : groups[i].id,
-                    label : groups[i].label,
-                    updatedAt : groups[i].updatedAt
-                }
-            }
-            resolve(resp);
-
-        });
-    });
+const getGroups = async(boardId) => {
+    try {
+        return await models.group.findAll({ where : { boardId : boardId } });
+    } catch(err) {
+        throw Error(err);
+    }
 }
 
-exports.addGroup = async(request) => {
+/**
+* Checks if a Teammember is a member of a certain board
+* @param {int} boardId id of the wanted board
+* @param2 {int} teammemberId id of the wanted teammember
+* @return {Boolean} True - if a member of the board, else False
+*/
+const isBoardTeammember = async(boardId, teammemberId) => {
+    const sequelize = await getSequelize();
+    const isMember = await sequelize.query('SELECT * FROM board_teammember WHERE "boardId" = (:boardId) AND "teammemberId" = (:teammemberId)',{
+        replacements : { boardId : boardId, teammemberId : teammemberId},
+        type: sequelize.QueryTypes.SELECT });
+    if(isMember.length == 0) {
```

Comment:
`if (!isMember)` is more clear than `if (isMember.length == 0)`.
By the way, try to use 3 times equal sign when going for conditions in JS as three times means strict comparison (type & value) whereas 2 times equal sign conditions mean only compare by value